sys/net/gnrc/pktshark/gnrc_pktshark.c

```diff
@@ -0,0 +1,721 @@
+            break;
+        default:
+            print_str(" opt=");
+            print_u32_dec(opt->type);
+        }
+    }
+}
+
+static void _dump_icmpv6_echo(const icmpv6_echo_t *echo, size_t payload_len)
+{
+    print_str(" seq=");
+    print_u32_dec(byteorder_ntohs(echo->seq));
+    print_str(" (");
+    print_u32_dec(payload_len);
+    print_str(" bytes)\n");
+}
+
+static void _dump_icmpv6(const icmpv6_hdr_t *hdr, size_t payload_len)
+{
+    print_str("\n\tICMPv6 ");
+
+    switch (hdr->type) {
+    case ICMPV6_DST_UNR:
+        print_str("Destination unreachable");
+        break;
+    case ICMPV6_PKT_TOO_BIG:
+        print_str("Packet too big");
+        break;
+    case ICMPV6_TIME_EXC:
+        print_str("Timeout");
+        break;
+    case ICMPV6_ECHO_REQ:
+        print_str("PING");
+        _dump_icmpv6_echo((icmpv6_echo_t *)hdr, payload_len - sizeof(icmpv6_echo_t));
+        break;
+    case ICMPV6_ECHO_REP:
+        print_str("PONG");
+        _dump_icmpv6_echo((icmpv6_echo_t *)hdr, payload_len - sizeof(icmpv6_echo_t));
+        break;
+    case ICMPV6_RTR_SOL:
+        _dump_rtr_sol((ndp_rtr_sol_t *)hdr, payload_len - sizeof(ndp_rtr_sol_t));
+        break;
+    case ICMPV6_RTR_ADV:
+        _dump_rtr_adv((ndp_rtr_adv_t *)hdr, payload_len - sizeof(ndp_rtr_adv_t));
+        break;
+    case ICMPV6_NBR_SOL:
+        _dump_nbr_sol((ndp_nbr_sol_t *)hdr, payload_len - sizeof(ndp_nbr_sol_t));
+        break;
+    case ICMPV6_NBR_ADV:
+        _dump_nbr_adv((ndp_nbr_adv_t *)hdr, payload_len - sizeof(ndp_nbr_adv_t));
+        break;
+    case 143:
+        print_str("Multicast Listener Report v2");
+        break;
+    default:
+        print_str("msg type ");
+        print_u32_dec(hdr->type);
+        print_str(" (");
+        print_u32_dec(payload_len);
+        print_str(" bytes)\n");
+    }
+    print_str("\n");
+}
+
+static void _print_coap_block(coap_pkt_t *pkt, uint16_t option)
+{
+    coap_block1_t block;
+    coap_get_block(pkt, &block, option);
+
+    if (option == COAP_OPT_BLOCK1) {
+        print_str(" B1[");
+    } else {
+        print_str(" B2[");
+    }
+    print_u32_dec(block.blknum);
+    print_str(".");
+    print_u32_dec(coap_szx2size(block.szx));
+
+    if (block.more) {
+        print_str(" M");
+    }
+
+    print_str("]");
+}
+
+static void _print_coap_format(unsigned format)
+{
+    print_str(" ");
+    switch (format) {
+    case COAP_FORMAT_TEXT:
+        print_str("text/plain");
+        break;
+    case COAP_FORMAT_LINK:
+        print_str("application/link-format");
+        break;
+    case COAP_FORMAT_JSON:
+        print_str("application/json");
+        break;
+    case COAP_FORMAT_CBOR:
+        print_str("application/cbor");
+        break;
+    case COAP_FORMAT_OCTET:
+        print_str("application/octet-stream");
+        break;
+    default:
+        print_str("fmt=");
+        print_u32_dec(format);
+        break;
+    }
+}
+
+static bool _dump_coap(const void *buf, size_t len)
```

Comment:
Not just here, but I think it would make sense to add each protocol to a submodule so users can pick and choose what they want to dissect.